1. Basics/task_3/hw3.php

```diff
@@ -0,0 +1,153 @@
+<?php
+date_default_timezone_set("Europe/Kyiv");
+
+/**
+ * @return string : input request sample as a string 
+ */
+function readHttpLikeInput() : string {
+    $f = fopen( 'php://stdin', 'r' );
+    $store = "";
+    $toread = 0;
+    while( $line = fgets( $f ) ) {
+        $store .= preg_replace("/\r/", "", $line);
+        if (preg_match('/Content-Length: (\d+)/',$line,$m)) 
+            $toread=$m[1]*1; 
+        if ($line == "\r\n") 
+              break;
+    }
+    if ($toread > 0) 
+        $store .= fread($f, $toread);
+    return $store;
+}
+
+$contents = readHttpLikeInput();
+
+/**
+ * @param int       $statuscode
+ * @param string    $statusmessage
+ * @param array     $headers
+ * @param string    $body
+ */
+function outputHttpResponse($statuscode, $statusmessage, $headers, $body) : void {
+    echo "HTTP/1.1 $statuscode $statusmessage\n";
+
+    foreach($headers as $header => $value) {         
+        echo "$header : $value\n";
+    }
+
+    echo "\n$body\n"; 
+}
+
+/**
+ * @param string    $method
+ * @param string    $uri
+ * @param array     $headers
+ * @param string    $body
+ */
+function processHttpRequest($method, $uri, $headers=null, $body=null) : void {
+    $statuscode = getResponseStatusCode($method, $uri);
```

Comment:
it's a good way to solve this task, but pay attention that you call `getResponseStatusMessage` twice. Refactor code, so there are no complications and duplicates

Reply:
Done.
Now function ```getResponseBody($statuscode, $statusmessage, $uri)``` takes 3 params and uses the second one instead of ```getResponseStatusMessage()``` calling.